magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,417 @@
+                        "conc_cog": conc_params.cog,
+                        "conc_core": conc_params.core,
+                        "conc_pixel": conc_params.pixel,
+                        "impact": impact,
+                        "pointing_alt": pointing_alt,
+                        "pointing_az": pointing_az,
+                        "alt": alt_rad,
+                        "az": az_rad,
+                        "wl": wl,
+                        "log_intensity": log_intensity,
+                        "timestamp": timestamp,
+                        "time_diff": time_diff,
+                        "zenith": zenith,
+                        "n_islands": n_islands,
+                        "n_pixels": n_pixels,
+                        "disp_dx": disp_dx,
+                        "disp_dy": disp_dy,
+                        "disp_norm": disp_norm,
+                        "disp_angle": disp_angle,
+                        "disp_sign": disp_sign,
+                        "src_x": src_x,
+                        "src_y": src_y,
+                    }
+                )
+
+    df = pd.DataFrame(data)
+    return df
+
+
+def main():
+    """Main function."""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--input_file",
+        "-i",
+        dest="input_file",
+        type=str,
+        required=True,
+        help="Path to an input .h5 DL1 data file",
+    )
+
+    parser.add_argument(
+        "--output_file",
+        "-o",
+        dest="output_file",
+        type=str,
+        default="./data",
+        help="Path to a directory where to save an output corrected DL1 file",
+    )
+
+    args = parser.parse_args()
+
+    subarray_info = SubarrayDescription.from_hdf(args.input_file)
+
+    tel_descriptions = subarray_info.tel
+    camgeom = {}
+    for telid, telescope in tel_descriptions.items():
+        camgeom[telid] = telescope.camera.geometry  # 1 - LSTCam; 2,3 - MAGICCam
+
+    optics_table = read_table(
+        args.input_file, "/configuration/instrument/telescope/optics"
+    )
+    focal_eq = {}
+    focal_eff = {}
+
+    for telid, telescope in tel_descriptions.items():
+        optics_row = optics_table[optics_table["optics_name"] == telescope.name]
+        if len(optics_row) > 0:
+            focal_length_eq = optics_row["equivalent_focal_length"][0]
+            focal_eq[telid] = focal_length_eq * u.m
+            focal_length_eff = optics_row["effective_focal_length"][0]
+            focal_eff[telid] = focal_length_eff * u.m
+
+    df_lst = process_telescope_data(
+        args.input_file, args.output_file, 1, focal_eq, focal_eff, camgeom
+    )
+    df_m1 = process_telescope_data(
+        args.input_file, args.output_file, 2, focal_eq, focal_eff, camgeom
+    )
+    df_m2 = process_telescope_data(
+        args.input_file, args.output_file, 3, focal_eq, focal_eff, camgeom
+    )
+
+    df_all = pd.concat([df_lst, df_m1, df_m2], ignore_index=True)
+
+    columns_to_convert = [
```

Comment:
all this code with the conversion also will not be needed if you write the way how I specified above